autocomplete_light/autocomplete/model.py

```diff
@@ -67,9 +67,16 @@ def order_choices(self, choices):
         Order choices using :py:attr:`order_by` option if it is set.
         """
         if self.values:
+            pk_name = "id"
+            try:
+                if len(choices) > 0:
+                    pk_name = choices[0]._meta.pk.name
+            except:
+                pass
+
             # Order in the user selection order when self.values is set.
-            clauses = ' '.join(['WHEN id=%s THEN %s' % (pk, i) for i, pk in
-                    enumerate(self.values)])
+            clauses = ' '.join(['WHEN %s="%s" THEN %s' % (pk_name, pk, i)
+                for i, pk in enumerate(self.values)])
             ordering = 'CASE %s END' % clauses
             return choices.extra(
                 select={'ordering': ordering}, order_by=('ordering',))
```

Comment:
I'm still not clear on whether or not having `self.values` means the queryset filter won't return anything else. If the queryset will usually return ['A', 'B', 'C', 'D', 'E'] because we have `order_by = 'name'`, and then we have `self.values = ['D', 'B'] I would expect the queryset to return ['D', 'B', A', 'C', "E']. That means that we need to respect the original ordering too. E.g.:

```python
return choices.extra(
                 select={'ordering': ordering}, order_by=('ordering',)+tuple(self.order_by))
```